File: src/main/java/com/bearbnb/service/PropertyAmenityService.java
package com.bearbnb.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.bearbnb.domain.PropertyAmenity;
import com.bearbnb.dto.PropertyAmenityDTO;
import com.bearbnb.dto.PropertyAmenitySearchDTO;
import com.bearbnb.dto.PropertyAmenityPageDTO;
import com.bearbnb.dto.PropertyAmenityConvertCriteriaDTO;
import com.bearbnb.service.GenericService;
import com.bearbnb.dto.common.RequestDTO;
import com.bearbnb.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PropertyAmenityService extends GenericService<PropertyAmenity, Integer> {

	List<PropertyAmenity> findAll();

	ResultDTO addPropertyAmenity(PropertyAmenityDTO propertyAmenityDTO, RequestDTO requestDTO);

	ResultDTO updatePropertyAmenity(PropertyAmenityDTO propertyAmenityDTO, RequestDTO requestDTO);

    Page<PropertyAmenity> getAllPropertyAmenitys(Pageable pageable);

    Page<PropertyAmenity> getAllPropertyAmenitys(Specification<PropertyAmenity> spec, Pageable pageable);

	ResponseEntity<PropertyAmenityPageDTO> getPropertyAmenitys(PropertyAmenitySearchDTO propertyAmenitySearchDTO);
	
	List<PropertyAmenityDTO> convertPropertyAmenitysToPropertyAmenityDTOs(List<PropertyAmenity> propertyAmenitys, PropertyAmenityConvertCriteriaDTO convertCriteria);

	PropertyAmenityDTO getPropertyAmenityDTOById(Integer propertyAmenityId);







}

File: bearbnb_ui/src/services/PropertyAmenityService.js
import http from "../http-common";

class PropertyAmenityService {
  getAllPropertyAmenitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/propertyAmenity/propertyAmenitys`, searchDTO);
  }

  get(propertyAmenityId) {
    return this.getRequest(`/propertyAmenity/${propertyAmenityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/propertyAmenity?field=${matchData}`, null);
  }

  addPropertyAmenity(data) {
    return http.post("/propertyAmenity/addPropertyAmenity", data);
  }

  update(data) {
  	return http.post("/propertyAmenity/updatePropertyAmenity", data);
  }
  
  uploadImage(data,propertyAmenityId) {
  	return http.postForm("/propertyAmenity/uploadImage/"+propertyAmenityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PropertyAmenityService();
